### ISSUE TITLE
self-update: -q doesn't quiet 'writing restic to'

### ISSUE THREAD

**Issue by jtagcat** [restic/restic#3350]
Output of `restic version`
--------------------------
restic 0.12.0 compiled with go1.15.8 on linux/amd64

Current
-------
```
$ restic self-udpate
writing restic to /home/restic/bin/restic
find latest release of restic at GitHub
restic is up to date
$

$ restic self-udpate -q
writing restic to /home/restic/bin/restic
$
```

Expected
--------
```
$ restic self-udpate -q
$
```

self-assign, deadline: 2022 :^)

**Comment by fd0** [comment]
Heh, good catch! Let me know if you want to work on it, should be easy to do :)

**Comment by kiinoda** [comment]
@fd0 if @jtagcat is busy or otherwise not interested, I'd be willing to work on it.

**Comment by fd0** [comment]
@kiinoda go ahead!

**Comment by kiinoda** [comment]
@fd0 since this is the first time that I'm looking at the restic code, I could use a bit of guidance as to what would be the proper way to fix this. The alternatives that I am considering are as follows:

* would it be better to test for `globalOptions.verbosity == 0` OR should I rather do this via `globalOptions.Quiet == true`? As I could see in the code, having `globalOptions.Quiet` set to `true` turns `globalOptions.verbosity` to `0` so we achieve the same thing BUT in context, one would be more suitable than the other. I tend to prefer testing for `globalOptions.Quiet == true` but I may be wrong.
* should I add the test above in `./cmd/restic/global.go` right inside the `Printf` definition or should I filter this straight where it is generated, in `./cmd/restic/cmd_self_update.go`? Putting it at the source achieves the same as putting it inside `global.go` BUT having the output suppressed in `cmd_self_update.go` filters this specific occurence while filtering output in `global.go` solves any other potential future issues such as this one, coming from any other "subsystem". And since `--quiet` means "really, keep quiet", I tend to prefer filtering inside `global.go`.

To sum it up, I would suggest testing for `globalOptions.Quiet == true` in `global.go`, in the definition of `Printf` and filter it there.

Please advise && let me know if I missed something. Thanks!

**Comment by jtagcat** [comment]
> would it be better to test for globalOptions.verbosity == 0 OR should I rather do this via globalOptions.Quiet == true? As I could see in the code, having globalOptions.Quiet set to true turns globalOptions.verbosity to 0 so we achieve the same thing BUT in context, one would be more suitable than the other. I tend to prefer testing for globalOptions.Quiet == true but I may be wrong.

Use `globalOptions.Quiet == true`. `globalOptions.verbosity` is most likely the toggle for extra verbosity (-v, --verbose).

> And since --quiet means "really, keep quiet",

Not really, with `--quiet`, errors should not be surpressed.  
I don't remember the code exactly, was there a separate channel for errors? If you're adding global tests, I'd suggest to make a separate PR (+ issue). Keep commits and PRs small, so they can be easily reviewed, changed, revised.
